dbux-projects/src/dataLib/PathwaysDataProvider.js

```diff
@@ -132,21 +135,27 @@ class StepCollection extends Collection {
     super('steps', pdp);
   }
 
-  handleEntryAdded(step) {
-    if (!step.stepGroupId) {
-      // convert group's string key to numerical id (since our indexes only accept numerical ids)
-      const {
-        type,
-        staticContextId
-      } = step;
+  postAddProcessed(steps) {
+    this.resolveStepGroupId(steps);
+  }
 
-      // const stepGroupKey = `${type}_${staticContextId}`;
-      const stepGroupKey = staticContextId ? `staticContextId_${staticContextId}` : `type_${type}`;
-      let stepGroupId = this.groupIdsByKey.get(stepGroupKey);
-      if (!stepGroupId) {
-        this.groupIdsByKey.set(stepGroupKey, stepGroupId = this.groupIdsByKey.size + 1);
+  resolveStepGroupId(steps) {
+    for (const step of steps) {
```

Comment:
Sorry, I forgot: Was our final decision not to use an `Index` for `groupIdsByKey`? If we had that index, we could also use it to look up `stepGroupId`.
If so, can we add a (hopefully, very short) comment on the function to explain why?

 My concern is that we are modifying "non-raw data" here, so at least we want to be very careful about how to deal with that.

Reply:
There are `stepsByGroupIndex` already, which can be used to look up `stepGroupId`. Not sure if `groupIdsByKey` still needs to be a index, it's a 1-1 map(i.e. `staticContextId_15` -> 10) that is used to resolve group id.